### PR TITLE
Ap motor clean style and fix some variable

### PR DESCRIPTION
**Pull request by khancyr**
tools use : clion ide, cpplint, cppcheck, clang-tidy

### CODE REVIEW

File: libraries/AP_Motors/AP_Motors6DOF.cpp [hunk 1]
```diff
@@ -197,22 +195,21 @@ int16_t AP_Motors6DOF::calc_thrust_to_pwm(float thrust_in) const
...
 {
-    int8_t i;
     int16_t motor_out[AP_MOTORS_MAX_NUM_MOTORS];    // final pwm values sent to the motor
 
     switch (_spool_mode) {
     case SHUT_DOWN:
         // sends minimum values out to the motors
         // set motor output based on thrust requests
-        for (i=0; i<AP_MOTORS_MAX_NUM_MOTORS; i++) {
+        for (int8_t i = 0; i < AP_MOTORS_MAX_NUM_MOTORS; i++) {
```
Comment (muramura) [inline libraries/AP_Motors/AP_Motors6DOF.cpp]: I think the variable "i" in this method is using multiple loops, so I think that it is OK as it is.

File: libraries/AP_Motors/AP_Motors6DOF.cpp [hunk 2]
```diff
@@ -354,23 +334,23 @@ void AP_Motors6DOF::output_armed_stabilizing_vectored()
...
+    for (int8_t i = 0; i < AP_MOTORS_MAX_NUM_MOTORS; i++) {
         if (motor_enabled[i]) {
             rpy_out[i] = roll_thrust * _roll_factor[i] +
                          pitch_thrust * _pitch_factor[i] +
                          yaw_thrust * _yaw_factor[i];
         }
     }
 
-    float forward_coupling_limit = 1-_forwardVerticalCouplingFactor*float(fabsf(throttle_thrust));
+    float forward_coupling_limit = 1-_forwardVerticalCouplingFactor * fabsf(throttle_thrust);
```
Comment (muramura) [inline libraries/AP_Motors/AP_Motors6DOF.cpp]: I do not know why whitespace is not added before and after the operator. I think that it is better to put it all if inserted.

File: libraries/AP_Motors/AP_MotorsHeli.h [hunk 3]
```diff
@@ -48,8 +48,8 @@ class AP_MotorsHeli : public AP_Motors {
 public:
 
     /// Constructor
-    AP_MotorsHeli( uint16_t         loop_rate,
-                   uint16_t         speed_hz = AP_MOTORS_HELI_SPEED_DEFAULT) :
+    AP_MotorsHeli(uint16_t         loop_rate,
```
Comment (muramura) [inline libraries/AP_Motors/AP_MotorsHeli.h]: I differ from other ways of having many variable types and variable white spaces. I think that it is better to combine with others. How is that?

File: libraries/AP_Motors/AP_MotorsSingle.cpp [hunk 4]
```diff
@@ -186,7 +178,7 @@ void AP_MotorsSingle::output_armed_stabilizing()
         }
     }
 
-    actuator_allowed = 1.0f - rp_scale * rp_thrust_max;
+    const float actuator_allowed = 1.0f - rp_scale * rp_thrust_max;     // amount of yaw we can fit in
```
Comment (muramura) [inline libraries/AP_Motors/AP_MotorsSingle.cpp]: I have not set the semi-colon and the number of whitespace of the beginning "//" of the comment to 2. I think that it is better to unify.

File: libraries/AP_Motors/AP_MotorsTri.cpp [hunk 5]
```diff
@@ -309,7 +303,7 @@ void AP_MotorsTri::thrust_compensation(void)
 {
     if (_thrust_compensation_callback) {
         // convert 3 thrust values into an array indexed by motor number
-        float thrust[4] { _thrust_right, _thrust_left, 0, _thrust_rear };
+        float thrust[4] { _thrust_right, _thrust_left, 0.0f, _thrust_rear };
```
Comment (muramura) [inline libraries/AP_Motors/AP_MotorsTri.cpp]: I do not understand the meaning of "{" after whitespace necessary," ("after the white space is unnecessary, I do not think that neither is unnecessary.

File: libraries/AP_Motors/AP_Motors_Class.h [hunk 6]
```diff
@@ -164,20 +164,25 @@ class AP_Motors {
     // set loop rate. Used to support loop rate as a parameter
     void                set_loop_rate(uint16_t loop_rate) { _loop_rate = loop_rate; }
 
-    enum pwm_type { PWM_TYPE_NORMAL=0, PWM_TYPE_ONESHOT=1, PWM_TYPE_ONESHOT125=2, PWM_TYPE_BRUSHED16kHz=3 };
+    enum pwm_type {
+        PWM_TYPE_NORMAL = 0,
```
Comment (muramura) [inline libraries/AP_Motors/AP_Motors_Class.h]: I like to have no column for constant specification. Please tell me about the judgment of not adjusting the column.

File: libraries/AP_Motors/examples/AP_Motors_test/AP_Motors_test.cpp [hunk 7]
```diff
@@ -111,7 +111,7 @@ void motor_order_test()
     hal.console->printf("testing motor order\n");
     motors.armed(true);
     for (int8_t i=1; i <= AP_MOTORS_MAX_NUM_MOTORS; i++) {
```
Comment (muramura) [inline libraries/AP_Motors/examples/AP_Motors_test/AP_Motors_test.cpp]: Why is there no white space in this state? I think that it is better to combine with others.

File: libraries/AP_Motors/AP_MotorsHeli.h (hunk 3 above)
Comment (khancyr) [inline libraries/AP_Motors/AP_MotorsHeli.h]: Yep it will be better

File: libraries/AP_Motors/AP_MotorsSingle.cpp (hunk 4 above)
Comment (khancyr) [inline libraries/AP_Motors/AP_MotorsSingle.cpp]: agreed

File: libraries/AP_Motors/AP_MotorsTri.cpp (hunk 5 above)
Comment (khancyr) [inline libraries/AP_Motors/AP_MotorsTri.cpp]: I just miss that one

File: libraries/AP_Motors/examples/AP_Motors_test/AP_Motors_test.cpp (hunk 7 above)
Comment (khancyr) [inline libraries/AP_Motors/examples/AP_Motors_test/AP_Motors_test.cpp]: agreed

File: libraries/AP_Motors/AP_Motors6DOF.cpp (hunk 2 above)
Comment (khancyr) [inline libraries/AP_Motors/AP_Motors6DOF.cpp]: miss that one

File: libraries/AP_Motors/AP_Motors6DOF.cpp (hunk 1 above)
Comment (khancyr) [inline libraries/AP_Motors/AP_Motors6DOF.cpp]: in C++ for simple type, I am pretty sure it is always a win to stay to classic for loop.

**Comment by khancyr** [comment]
@muramura thanks for looking this !

**Comment by khancyr** [comment]
I forget to make it specific with a commit but I am pretty sure that there is a bug in AP_motorsSingle : https://github.com/ArduPilot/ardupilot/pull/6440/commits/982804e80dbe4819821075fc72fc14f8bc7eb1e6#diff-bd678d85e304e05319eb32eeedd00ebfR221

File: libraries/AP_Motors/AP_MotorsMulticopter.h [hunk 8]
```diff
@@ -123,7 +123,7 @@ class AP_MotorsMulticopter : public AP_Motors {
     float               get_compensation_gain() const;
 
     // convert thrust (0~1) range back to pwm range
-    int16_t             calc_thrust_to_pwm(float thrust_in) const;
+    virtual int16_t     calc_thrust_to_pwm(float thrust_in) const;
```
Comment (OXINARF) [inline libraries/AP_Motors/AP_MotorsMulticopter.h]: Then add the override keyword in 6DOF.

**Comment by khancyr** [comment]
closed add too old

**Pull request closed without merging**